### ISSUE TITLE
Toggle Icon Visual Components

### ISSUE THREAD

**Issue by Brittank88** [absoIute/Mega-Hack-Pro-Future#416]
Rolling #414 and #415 into a single issue that accounts for them both.

Example usages:

- Hide UFO dome.
- Hide mini player inside vehicles where it is visible (UFO, ship).
- Floating robot? Lmao

**Comment by Nicko2102** [comment]
> * Floating robot? Lmao

yes please, removing robot/spider legs was gonna be my next post lmao

**Comment by Brittank88** [comment]
In future, try your best to roll issues into one if they cover basically the same overall feature!

Otherwise, these sound pretty fun.

**Comment by Nicko2102** [comment]
> In future, try your best to roll issues into one if they cover basically the same overall feature!

I will, thanks for the heads up :)

**Comment by Nicko2102** [comment]
I was also going to add (as another visual component to toggle)(though i don't know how well it will work), the low opacity overlay things that don't really have a colour (example shown is the dot in the spider eye, it also appears on the default spider and some icons but i use a texture pack)
![image](https://user-images.githubusercontent.com/83476899/151290545-8fb2a1c0-4eae-4d38-8faa-60e5bf7a7822.png)

**Comment by Nicko2102** [comment]
more examples are the strip in the middle of this wave (still shows up when the whole thing is black)
![image](https://user-images.githubusercontent.com/83476899/151290871-ba0d541f-9d45-4b56-9200-9a692eeffcf4.png)
and just a better view of the spider eye
![image](https://user-images.githubusercontent.com/83476899/151290974-2111926d-8250-433f-8347-57fe9a5005bc.png)
hopefully i explained it ok enough that you can tell what i'm getting at

**Comment by Brittank88** [comment]
Makes sense, though I wonder at what point this is just better covered by a texture pack.

**Comment by Brittank88** [comment]
> Makes sense, though I wonder at what point this is just better covered by a texture pack.

I think the point at which this stops being a MH feature is when it's no longer a separate component in the gamesheet.

**Comment by Nicko2102** [comment]
that's fair, let me see

**Comment by Nicko2102** [comment]
a bit of research later, this is what i found
for example, this spider
![image](https://user-images.githubusercontent.com/83476899/151293645-befeeb42-139e-45f1-b148-5d1dba21325c.png)
is made up of three components (not including legs)
1-Colour 1
![image](https://user-images.githubusercontent.com/83476899/151293387-c7089ad8-ad67-4dec-9917-4d02133026ca.png)
2-Colour 2
![image](https://user-images.githubusercontent.com/83476899/151293451-ca4da6e6-6374-4a06-bab8-16acba97ceb2.png)
3-Extra
![image](https://user-images.githubusercontent.com/83476899/151293529-7b97170b-2b26-4504-8e82-6dc86113df67.png)

Also, the pList for this icon has 3 parts, which correspond to each of these gamesheet components
![image](https://user-images.githubusercontent.com/83476899/151293631-8994ac6e-0773-463f-92eb-a280210d871e.png)

**Comment by Nicko2102** [comment]
might be good to make this compatible with TxtrLdr as well

**Comment by Brittank88** [comment]
Good on you for doing your own research! Wonderful!

**Comment by totallyphysicallypossible** [comment]
The way that icons are made in the plist is kinda weird ngl
UFO Dome uses bird_‌##‌_3_001
Extra stuff is basically above me
Robots and Spiders are made with 4 parts, each having 3
spider_‌##‌_01_001 (the main thing)
spider_‌##‌_02_001 (legs)
spider_‌##‌_03_001 (back legs)
spider_‌##‌_04_001 (thing that i cant explain but you most likely wouldn't notice if you deleted)

I can see No UFO Dome and floating robots and spiders being somewhat used, though I'm not sure about no extra as it's used in a lot of texture packs, some even have the outline itself in the extra and 1 and 2 just used for primary and secondary

**Comment by Nicko2102** [comment]
> I'm not sure about no extra as it's used in a lot of texture packs, some even have the outline itself in the extra and 1 and 2 just used for primary and secondary

Fair enough, but my guess is that these will be separate options (i.e. floating robot, no ufo dome etc will be separate toggles so you can have one without the other), so if it makes a texture pack look bad you can always just not use that option

**Comment by Nicko2102** [comment]
> Robots and Spiders are made with 4 parts, each having 3 spider_‌##‌_01_001 (the main thing) spider_‌##‌_02_001 (legs) spider_‌##‌_03_001 (back legs) spider_‌##‌_04_001 (thing that i cant explain but you most likely wouldn't notice if you deleted)

Not sure how related this is (what I'm about to say, not your comment), but i noticed something weird... so the spider's front legs have 3 parts, one for col1, one for col2, and one for glow, but if you look at the spider's front legs, you can see none of it is col2. So i looked at what the plist defines as the col2 part and...
![image](https://user-images.githubusercontent.com/83476899/151367407-791227d8-2c99-49b2-9be3-76c3341aeebe.png)
it's one of these tiny things. a 6x4 tiny rectangle. tbh, it might just be using the empty space between them, which just gave me another idea...

if it isn't possible to get rid of specific parts of the plist, how about just setting the coords of the png to an empty part of the plist, since basically the entire bottom of the png is blank?

**Comment by totallyphysicallypossible** [comment]
Pointing the plist to an empty part of the png would work, but I'm not sure how it would know which parts are empty but I mean it can just use 99999,99999 and it'll work

Using a texturesize of 1,1 is preferable to make sure that something doesn't get in and make it look all weird

I'm not sure if using texturesize 0,0 works but someone can test it

**Comment by Nicko2102** [comment]
for gamesheet2 (which contains all the icons), using any part of the general bottom area would work for all 3 resolution levels, if you want proof i can show in a bit

**Comment by totallyphysicallypossible** [comment]
I've made texture packs for a bit and yeah, but some (or most) texture packs use the bottom area as free space to put icons so I don't really know about that

**Comment by Nicko2102** [comment]
> but some (or most) texture packs use the bottom area as free space to put icons so I don't really know about that

Oh that's a good point, didn't even think about that

**Comment by absoIute** [comment]
YES

will add options for
- hiding player in ship/ufo (big or mini)
- hiding the ship/ufo vehicle 
- hiding the UFO dome
- those legs

**Comment by zohnannor** [comment]
I already see tons of videos on YT with the names like "So I tried MegaHack's Toggle Icon Visual Components feature..."

**Comment by Nicko2102** [comment]
> * hiding the ship/ufo vehicle

Is this just the opposite of the above one? like you can see the icon jumping/flying but not the vehicle? if so that's hilarious

**Comment by Nicko2102** [comment]
> I already see tons of videos on YT with the names like "So I tried MegaHack's Toggle Icon Visual Components feature..."

aeonair

**Comment by Nicko2102** [comment]
not sure if you were gonna do this already but maybe have the feet and legs (for robot/spider) separate options, because i would LOVE to have a robot with feet and no legs running around like this
![image](https://user-images.githubusercontent.com/83476899/151613479-f25afdff-f793-459a-a63a-a453e1b3e217.png)

**Comment by absoIute** [comment]
Might do that too we'll see!

Hitboxes will be the same though, so I might play around with changing sprite sizes to make them not feel weird

**Comment by Brittank88** [comment]
> bird_‌##‌_3_001

**bird**

That's a new one, lmao. Didn't know about that.

**Comment by Brittank88** [comment]
> not sure if you were gonna do this already but maybe have the feet and legs (for robot/spider) separate options, because i would LOVE to have a robot with feet and no legs running around like this ![image](https://user-images.githubusercontent.com/83476899/151613479-f25afdff-f793-459a-a63a-a453e1b3e217.png) ^garbage art^

RAYMAN ROBOT
